File: src/main/java/com/praxis/topics/entity/Topic.java
package com.praxis.topics.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "Topics")
public class Topic {
    @Id
    private String name;
    //private String id;
    private String description;
    private int status;
    private int teachers_count;
    private int students_count;
    private String createdAt;
    private String openedAt;
    private String closedAt;
    private String expiredAt;

    public Topic(String name, String description) {
        this.name = name;
        this.description = description;
    }

    public Topic() {

    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public int getTeachers_count() {
        return teachers_count;
    }

    public void setTeachers_count(int teachers_count) {
        this.teachers_count = teachers_count;
    }

    public int getStudents_count() {
        return students_count;
    }

    public void setStudents_count(int students_count) {
        this.students_count = students_count;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getOpenedAt() {
        return openedAt;
    }

    public void setOpenedAt(String openedAt) {
        this.openedAt = openedAt;
    }

    public String getClosedAt() {
        return closedAt;
    }

    public void setClosedAt(String closedAt) {
        this.closedAt = closedAt;
    }

    public String getExpiredAt() {
        return expiredAt;
    }

    public void setExpiredAt(String expiredAt) {
        this.expiredAt = expiredAt;
    }
}

File: src/app/shared/topic.ts
import { TopicRequest  } from './topicRequest';
export class Topic {
    id:string;
    name: string;
    description: string;
    createdAt: string;
    expiredAt: any;
    openedAt: any;
    teachers: number;
    students: number;
    constructor(topic?: TopicRequest){
        if(topic){
            this.id = topic.id;
            this.name = topic.name;
            this.description = topic.description;
            this.createdAt = topic.createdAt;
            this.expiredAt = topic.expiredAt;
            this.openedAt = topic.openedAt;
            this.teachers = topic.teachers;
            this.students = topic.students;
        }else{
            this.teachers = 0;
            this.students = 0;
        }
    }
}
